install.hs

```diff
@@ -198,36 +193,30 @@ cabalInstallHie versionNumber = do
             (localBin </> "hie-" ++ dropExtension versionNumber <.> exe)
 
 cabalBuildDoc :: Action ()
-cabalBuildDoc = generateHoogleDatabase $ do 
-  localBin <- getLocalBin
-  execCabal_ ["new-install", "--symlink-bindir=" ++ localBin, "hoogle"]
+cabalBuildDoc = do
+  execCabal_ ["new-build", "hoogle", "generate"]
   execCabal_ ["new-exec", "hoogle", "generate"]
 
-generateHoogleDatabase :: Action () -> Action ()
-generateHoogleDatabase installIfNecessary = do
-  mayHoogle <- liftIO $ findExecutable "hoogle"
-  case mayHoogle of
-    Nothing -> installIfNecessary
-    Just hoogle -> command_ [] "hoogle" ["generate"]
-
-
 cabalTest :: VersionNumber -> Action ()
 cabalTest versionNumber = do
   configureCabal versionNumber
   execCabal_ ["new-test"]
 
-installCabal :: GhcPath -> Action ()
-installCabal ghc = do
-  execStack_ ["install", "--stack-yaml=shake.yaml", "cabal-install"]
+installCabal :: Action ()
+installCabal = do
+  -- install `cabal-install` if not already installed
+  unlessM (existsExecutable "cabal") $ do
```

Comment:
Requires a version check, what if the found cabal version is something like `1.2.0.0`? Do we know whether hie could handle that.?